javalib/src/main/scala/java/lang/Math.scala

```diff
@@ -1,324 +1,347 @@
+    `llvm.exp.f64`(a)
+
+  @inline def expm1(a: scala.Double): scala.Double =
+    cmath.expm1(a)
+
+  @inline def floor(a: scala.Double): scala.Double =
+    `llvm.floor.f64`(a)
+
+  def floorDiv(a: scala.Int, b: scala.Int): scala.Int = {
+    val quot = a / b
+    if ((a < 0) == (b < 0) || quot * b == a) quot
+    else quot - 1
   }
 
-  def hypot(a: scala.Double, b: scala.Double): scala.Double = {
-    // http://en.wikipedia.org/wiki/Hypot#Implementation
-    if (abs(a) == scala.Double.PositiveInfinity ||
-        abs(b) == scala.Double.PositiveInfinity)
-      scala.Double.PositiveInfinity
-    else if (a.isNaN || b.isNaN)
-      scala.Double.NaN
-    else if (a == 0 && b == 0)
-      0.0
-    else {
-      //To Avoid Overflow and UnderFlow
-      // calculate |x| * sqrt(1 - (y/x)^2) instead of sqrt(x^2 + y^2)
-      val x = abs(a)
-      val y = abs(b)
-      val m = max(x, y)
-      val t = min(x, y) / m
-      m * sqrt(1 + t * t)
-    }
+  def floorDiv(a: scala.Long, b: scala.Long): scala.Long = {
+    val quot = a / b
+    if ((a < 0) == (b < 0) || quot * b == a) quot
+    else quot - 1
   }
 
-  def expm1(a: scala.Double): scala.Double = {
-    // https://github.com/ghewgill/picomath/blob/master/javascript/expm1.js
-    if (a == 0 || a.isNaN)
-      a
-    // Power Series http://en.wikipedia.org/wiki/Power_series
-    // for small values of a, exp(a) = 1 + a + (a*a)/2
-    else if (abs(a) < 1E-5)
-      a + 0.5 * a * a
-    else
-      exp(a) - 1.0
+  def floorMod(a: scala.Int, b: scala.Int): scala.Int = {
+    val rem = a % b
+    if ((a < 0) == (b < 0) || rem == 0) rem
+    else rem + b
   }
 
-  def sinh(a: scala.Double): scala.Double = {
-    if (a.isNaN || a == 0.0 || abs(a) == scala.Double.PositiveInfinity)
-      a
-    else
-      (exp(a) - exp(-a)) / 2.0
+  def floorMod(a: scala.Long, b: scala.Long): scala.Long = {
+    val rem = a % b
+    if ((a < 0) == (b < 0) || rem == 0) rem
+    else rem + b
   }
 
-  def cosh(a: scala.Double): scala.Double = {
-    if (a.isNaN)
-      a
-    else if (a == 0.0)
-      1.0
-    else if (abs(a) == scala.Double.PositiveInfinity)
-      scala.Double.PositiveInfinity
-    else
-      (exp(a) + exp(-a)) / 2.0
+  @inline def getExponent(a: scala.Float): scala.Int =
+    cmath.ilogbf(a)
+
+  @inline def getExponent(a: scala.Double): scala.Long =
+    cmath.ilogb(a)
+
+  @inline def hypot(a: scala.Double, b: scala.Double): scala.Double =
+    cmath.hypot(a, b)
+
+  @inline def IEEEremainder(f1: scala.Double, f2: scala.Double): Double =
+    cmath.remainder(f1, f2)
+
+  @inline def incrementExact(a: scala.Int): scala.Int =
+    addExact(a, 1)
+
+  @inline def incrementExact(a: scala.Long): scala.Long =
+    addExact(a, 1L)
+
+  @inline def log(a: scala.Double): scala.Double =
+    `llvm.log.f64`(a)
+
+  @inline def log10(a: scala.Double): scala.Double =
+    `llvm.log10.f64`(a)
+
+  @inline def log1p(a: scala.Double): scala.Double =
+    log(a + 1)
```

Comment:
There seems to exist an implementation of `log1p` in LLVM: https://github.com/chapuni/llvm-project/blob/master/libclc/generic/lib/math/log1p.cl
Maybe not a built-in, but worth looking into whether you can call that.

`log(a + 1)` can suffer from imprecisions wrt. its documented guarantees. (and yes, I know the Scala.js implementation also uses `log(a + 1)`, and it's not quite right).